pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -767,47 +768,38 @@ func (s *FlinkStateMachine) handleSubmittingJob(ctx context.Context, app *v1beta
 	if job == nil {
 		return statusUnchanged, errors.Errorf("Could not find job %s", s.flinkController.GetLatestJobID(ctx, app))
 	}
-
-	jobStartTime := getJobStartTimeInUTC(job.StartTime)
-	now := time.Now().UTC()
 	cfg := config.GetConfig()
-	logger.Info(ctx, "Job vertex timeout config is ", cfg.FlinkJobVertexTimeout)
 	flinkJobVertexTimeout := cfg.FlinkJobVertexTimeout
-	if now.Before(jobStartTime.Add(flinkJobVertexTimeout.Duration)) {
-		allVerticesRunning, hasFailure, failedVertexIndex := monitorAllVerticesState(job)
-
-		// fail fast
-		if hasFailure {
-			s.flinkController.LogEvent(ctx, app, corev1.EventTypeWarning, "JobRunningFailed",
-				fmt.Sprintf(
-					"Vertex %d with name [%s] state is Failed", failedVertexIndex, job.Vertices[failedVertexIndex].Name))
-			return s.deployFailed(app)
-		}
-		return updateJobAndReturn(ctx, job, s, allVerticesRunning, app, hash)
+	logger.Info(ctx, "Monitoring job vertices with timeout ", flinkJobVertexTimeout)
+	jobStarted, err := monitorJobStart(job, flinkJobVertexTimeout)
```

Comment:
nit: would be good to call the method name `monitorJobSubmission` and `jobStarted` to `status`

Reply:
There are **only** two hard things in Computer Science: cache invalidation and naming things :)  
My rationale to rename `jobStarted` to `status` was primarily due my other suggestion for renaming the method to `monitorJobSubmission`.  Given the method was returning a boolean, a status felt more natural.  Okay to keep it as `jobStarted`.